src/proxy/moralis.rs

```diff
@@ -0,0 +1,330 @@
+                log_format!(
+                    remote_addr.ip(),
+                    payload.signed_message.address,
+                    req.uri(),
+                    "Error inserting JWT into HTTP header: {}, returning 500.",
+                    e
+                )
+            );
+            return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    let original_req_uri = req.uri().clone();
+
+    if let Err(e) = modify_request_uri(&mut req, &payload, proxy_route).await {
+        log::error!(
+            "{}",
+            log_format!(
+                remote_addr.ip(),
+                payload.signed_message.address,
+                original_req_uri,
+                "Error modifying request Uri: {}, returning 500.",
+                e
+            )
+        );
+        return response_by_status(StatusCode::INTERNAL_SERVER_ERROR);
+    }
+
+    let additional_headers = &[
+        header::CONTENT_LENGTH,
+        HeaderName::from_bytes(X_AUTH_PAYLOAD.as_bytes())?,
+    ];
+    remove_unnecessary_headers(&mut req, additional_headers)?;
+
+    req.headers_mut()
+        .insert(HeaderName::from_static(X_FORWARDED_FOR), x_forwarded_for);
+    req.headers_mut()
+        .insert(header::ACCEPT, APPLICATION_JSON.parse()?);
+
+    let https = HttpsConnector::new();
+    let client = hyper::Client::builder().build(https);
+
+    let target_uri = req.uri().clone();
+    let res = match client.request(req).await {
+        Ok(t) => t,
+        Err(e) => {
+            log::warn!(
+                "{}",
+                log_format!(
+                    remote_addr.ip(),
+                    payload.signed_message.address,
+                    original_req_uri,
+                    "Couldn't reach {}: {}. Returning 503.",
+                    target_uri,
+                    e
+                )
+            );
+            return response_by_status(StatusCode::SERVICE_UNAVAILABLE);
+        }
+    };
+
+    Ok(res)
+}
+
+/// Modifies the URI of an HTTP request by replacing it to outbound URI specified in `ProxyRoute`,
+/// while incorporating the path and query parameters from the payload's URI.
+async fn modify_request_uri(
+    req: &mut Request<Body>,
+    payload: &MoralisPayload,
+    proxy_route: &ProxyRoute,
+) -> GenericResult<()> {
+    let mut proxy_outbound_parts = proxy_route.outbound_route.parse::<Uri>()?.into_parts();
+
+    let payload_uri: Uri = payload.uri.as_str().parse()?;
+
+    let path_and_query =
+        PathAndQuery::from_str(payload_uri.path_and_query().map_or("/", |pq| pq.as_str()))?;
+    // Append the path and query from the payload URI to the proxy outbound URI.
+    proxy_outbound_parts.path_and_query = Some(path_and_query);
+
+    // Reconstruct the full URI with the updated parts.
+    let new_uri = Uri::from_parts(proxy_outbound_parts)?;
+
+    // Update the request URI.
+    *req.uri_mut() = new_uri;
+    Ok(())
+}
+
+pub(crate) async fn validation_middleware_moralis(
```

Comment:
This should be documented in the README file (currently we only explain how quicknode middleware works).

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-proxy/pull/22/commits/c639542afc548a1ce9423dbc54584f07149b7fa0